### ISSUE TITLE
[Feature request] Add option to define running work directory

### ISSUE THREAD

**Issue by Diogo-Rossi** [taskipy/taskipy#81]
If I have a task depending on a directory, as in the docs:

```
[tool.taskipy.tasks]
test = "python -m unittest tests/test_*.py"
```

If I run the command `task test` in a folder other than the project's root directory (lets say `scr/`)
it will raise an error, since the folder `test/` won't be found.

This feature would allow to define a predefined work directory for the task.

It could be relative to the `pyproject.toml`'s directory, maybe?

```
[tool.taskipy.settings]
cwd = "."
```

Is it feasible?

Thank you very much for this tool!

**Comment by carl0smat3us** [comment]
> If I have a task depending on a directory, as in the docs:
> 
> ```
> [tool.taskipy.tasks]
> test = "python -m unittest tests/test_*.py"
> ```
> 
> If I run the command `task test` in a folder other than the project's root directory (lets say `scr/`) it will raise an error, since the folder `test/` won't be found.
> 
> This feature would allow to define a predefined work directory for the task.
> 
> It could be relative to the `pyproject.toml`'s directory, maybe?
> 
> ```
> [tool.taskipy.settings]
> cwd = "."
> ```
> 
> Is it feasible?
> 
> Thank you very much for this tool!

I'm unsure about this feature because it can often be indispensable. Frequently, we need to use Taskipy in various directories such as scripts, src, tests, docs, etc. This largely depends on the project. For instance, if you aim to avoid repeating the same directory multiple times, you can utilize variables.

For example:

```toml
[tool.taskipy.variables]
path = "path/to/my_module"

[tool.taskipy.tasks]
lint = { cmd = "pylint {path}", use_vars = true }
black = { cmd = "black {path}", use_vars = true }
```

**Comment by Diogo-Rossi** [comment]
Thanks for the reply!

The thing is: Can `"path/to/my_module"` be relative?
If not, I could not share this setting in a public repo. Am I right?

**Comment by carl0smat3us** [comment]
> Thanks for the reply!
> 
> The thing is: Can `"path/to/my_module"` be relative, lets say, to `pyproject.toml`?
> 
> If not, I can not share that setting in a public repo. Am I right?
> 
> 

Yeah the path It's always relative to your pyproject.toml file. 

I mean where you declare the pyproject.toml file will be always your work directory.

**Comment by Diogo-Rossi** [comment]
> Yeah the path It's always relative to your pyproject.toml file.
> 
> I mean where you declare the pyproject.toml file will be always your work directory.

Sorry but, can you clarify? That is not what is happening here:

```
./
│ -  pyproject.toml
│
├───src/
└───tests/
        - myFile.txt
```

1. I have this file `myFile.txt` which content is "Hello"

2. And I have  a `pyproject.toml` with the task:
```toml
[tool.taskipy.tasks]
test = "cat tests/myfile.txt"
```

It works ok when I'm in the root dir:
```
❯ task test
Hello!
```
But not when I in `src`, for instance:
```
❯ cd src
❯ task test
cat: ./tests/myfile.txt: No such file or directory
```

**Comment by carl0smat3us** [comment]
@illBeRoy

**Comment by Diogo-Rossi** [comment]
@illBeRoy Thanks for considering!

> What do you think? Would you like to contribute this feature?

I actually think that will be easy, because you already foresaw an argument `cwd` in this function:

https://github.com/taskipy/taskipy/blob/024933d703fb490c8eccbf5e10722d8bc29e20a9/taskipy/cli.py#L16-L17

---

But I found some issues in installing dependencies for the project.

By running `poetry install` (with and without the `.lock` file) I ended up with the following error:

```bash
  at ~\AppData\Roaming\pypoetry\venv\Lib\site-packages\poetry\installation\chef.py:164 in _prepare
      160│
      161│                 error = ChefBuildError("\n\n".join(message_parts))
      162│
      163│             if error is not None:
    → 164│                 raise error from None
      165│
      166│             return path
      167│
      168│     def _prepare_sdist(self, archive: Path, destination: Path | None = None) -> Path:

Note: This error originates from the build backend, and is likely not a problem with poetry but with wrapt (1.13.3) not supporting PEP 517 builds. You can verify this by running 'pip 
wheel --no-cache-dir --use-pep517 "wrapt (==1.13.3)"'.
```

Do you know what am I doing wrong?

**Comment by illBeRoy** [comment]
I believe there's an issue with python 3.12 and one of the dev dependencies. I'll look into this

**Comment by illBeRoy** [comment]
seems like we needed to update pylint. I'm fixing this on #82

**Comment by illBeRoy** [comment]
solved. please pull master and try reinstalling

**Comment by Diogo-Rossi** [comment]
It's working now, thanks @illBeRoy!

I'll work on the feature

**Comment by Diogo-Rossi** [comment]
I ended up using another approach, since the `run` functions calls an internal object to manipulate tasks.
I created a new method:
https://github.com/taskipy/taskipy/pull/83/files#diff-10fcc779fd0cc5d4e11ffea1157dcd3bb4a7282fe8ae915d26b4227ef9181c4fR198-R210

**Comment by illBeRoy** [comment]
Merged and released on 1.14.0. Thank you for your contribution!
